doc/adrs/095_simplify_the_manipulation_of_nodes_by_switching_to_a_common_box_model.adoc

```diff
@@ -0,0 +1,458 @@
+:imagesdir: images/095
+= ADR-095 - Simplify the manipulation of nodes by switching to a common box model
+
+== Context
+
+When the project was first started a couple years ago, we only used ELK and a fully automatic layout.
+After that, we wanted to let end users move nodes by themselves and so we introduced the incremental layout.
+
+This incremental layout works by having an existing layout and the current event in order to compute a new layout.
+It was a very simple approach which produced good results for a low cost.
+
+It does not really know how to layout everything from scratch, which is why everything is layouted with ELK first, it simply determines how an event impact the current layout.
+One of the issue with this approach is that since we have more and more complex diagrams, we need to handle a very complex set of interactions between an existing layout and all the possible events.
+On top of that, we also had to deal with complex unsynchronized diagrams which can start with a blank state.
+In those case, ELK will just layout an empty diagram and the incremental layout will have to handle everything manually.
```

Comment:
"manually" => "by itself".

It's not *that* ambiguous in the context, but "manually" could be interpreted as "the end-user has to layout every element by himself".